test/extended/networking/route_advertisements.go

```diff
@@ -0,0 +1,450 @@
+			o.Expect(len(workerNodesOrderedNames)).Should(o.BeNumerically(">", 1))
+			externalNodeName = workerNodesOrderedNames[0]
+			advertisedPodsNodes = workerNodesOrderedNames[1:]
+
+			g.By("Determining the interface that will be used for packet sniffing")
+			packetSnifferInterface, err = findPacketSnifferInterface(oc, networkPlugin, []string{externalNodeName})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			framework.Logf("Using interface %s for packet captures", packetSnifferInterface)
+
+			g.By("Turn on the BGP advertisement of PodNetwork")
+			_, err = runOcWithRetry(oc.AsAdmin(), "patch", "ra", "default", "--type=merge", `-p={"spec":{"advertisements":["PodNetwork"]}}`)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("Makes sure that the RouteAdvertisements is accepted")
+			waitErr := wait.PollUntilContextTimeout(context.TODO(), 10*time.Second, 30*time.Second, true, func(context.Context) (bool, error) {
+				ra, err := getRouteAdvertisements(oc, "default")
+				if err != nil {
+					return false, err
+				}
+				condition := meta.FindStatusCondition(ra.Status.Conditions, "Accepted")
+				return condition.Status == metav1.ConditionTrue, nil
+			})
+			o.Expect(waitErr).NotTo(o.HaveOccurred())
+
+			g.By("Makes sure the FRR configuration is generated for each node")
+			for _, nodeName := range workerNodesOrderedNames {
+				frr, err := getGeneratedFrrConfigurationForNode(oc, nodeName)
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(frr).NotTo(o.BeNil())
+			}
+
+			g.By("Spawning the packet sniffer pods on the workers")
+			packetSnifferDaemonSet, err = createPacketSnifferDaemonSet(oc, externalNamespace, advertisedPodsNodes, targetProtocol, targetPort, packetSnifferInterface)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("Creating the test target deployment with number of advertisedPodsNodes")
+			ingressDomain, err = getIngressDomain(oc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			deployName, routeName, err = createAgnhostDeploymentAndIngressRoute(oc, targetNamespace, "", ingressDomain, len(advertisedPodsNodes), advertisedPodsNodes)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			podList, err = clientset.CoreV1().Pods(targetNamespace).List(context.TODO(), metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", deployName)})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(len(podList.Items)).To(o.Equal(len(advertisedPodsNodes)))
+
+			g.By("Retrieving the PodIPs of the test source deployment")
+			v4PodIPSet = make(map[string]string)
+			v6PodIPSet = make(map[string]string)
+			for _, pod := range podList.Items {
+				for _, ip := range pod.Status.PodIPs {
+					IP := net.ParseIP(ip.IP)
+					o.Expect(IP).NotTo(o.BeNil())
+					if IP.To4() != nil {
+						v4PodIPSet[ip.IP] = pod.Spec.NodeName
+					} else {
+						v6PodIPSet[ip.IP] = pod.Spec.NodeName
+					}
+				}
+			}
+		})
+
+		g.It("pods should communicate with external host without being SNATed", func() {
```

Comment:
What route of the gateway router/node is the pod exercising when reaching out to the external host?
I ask because it would be nice if this were a route imported with BGP instead of the default route which would give us coverage fro import logic as well. Unless you have additional test cases planned for that.

Reply:
I'll add the route to a real external host in the BGP setup script.